### PR TITLE
fix(dids): allow to pass in a key instance when creating a did document

### PR DESCRIPTION
**Pull request by berendsliedrecht**
Signed-off-by: Berend Sliedrecht <sliedrecht@berend.io>

### CODE REVIEW

File: packages/core/src/modules/dids/methods/peer/PeerDidRegistrar.ts [hunk 1]
```diff
@@ -37,23 +37,27 @@ export class PeerDidRegistrar implements DidRegistrar {
...
+
+        if (!key && keyType) {
+          key = await agentContext.wallet.createKey({
+            keyType,
+            seed,
+            privateKey,
+          })
+        }
+
+        if (!key) {
```
Comment (TimoGlastra) [inline packages/core/src/modules/dids/methods/peer/PeerDidRegistrar.ts:50]: need to handle where both keyType and key are passed, but they don't match

Comment (TimoGlastra) [reply packages/core/src/modules/dids/methods/peer/PeerDidRegistrar.ts:50]: also need to handle the case where privateKey/seed is passed and key as well. That should cause an error. As it results in ambigious behaviour (I have no idea without looking at the code which will take precedence)

Comment (TimoGlastra) [reply packages/core/src/modules/dids/methods/peer/PeerDidRegistrar.ts:50]: (same for all files)

**Review by TimoGlastra — CHANGES_REQUESTED** [review]

**Comment by TimoGlastra** [comment]
can you also add a changest?

**Comment by berendsliedrecht** [comment]
> can you also add a changest?

how?

**Comment by TimoGlastra** [comment]
image

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**